lib/bulk.go

```diff
@@ -208,18 +208,28 @@ func (f *Force) GetBatchInfo(jobId string, batchId string) (result BatchInfo, er
 
 func (f *Force) GetBatches(jobId string) (result []BatchInfo, err error) {
 	url := fmt.Sprintf("%s/services/async/%s/job/%s/batch", f.Credentials.InstanceUrl, apiVersionNumber, jobId)
-	body, _, err := f.httpGetBulk(url)
+	body, contentType, err := f.httpGetBulk(url)
 
 	var batchInfoList struct {
-		BatchInfos []BatchInfo `xml:"batchInfo"`
+		BatchInfos []BatchInfo `xml:"batchInfo" json:"batchInfo"`
 	}
 
-	xml.Unmarshal(body, &batchInfoList)
-	result = batchInfoList.BatchInfos
-	if len(result) == 0 {
-		var fault LoginFault
-		xml.Unmarshal(body, &fault)
-		err = errors.New(fmt.Sprintf("%s: %s", fault.ExceptionCode, fault.ExceptionMessage))
+	if contentType == "JSON" {
+		json.Unmarshal(body, &batchInfoList)
+		result = batchInfoList.BatchInfos
+		if len(result) == 0 {
+			var fault LoginFault
+			json.Unmarshal(body, &fault)
```

Comment:
I'm not sure if this is a valid scenario.  If it's invalid job id, for example, we'll get back an XML response so no JSON job is found.

Reply:
I can't find a case for behaviour either. The same snippet seems to appear on all Get* methods however. Should I clean it up?

Reply:
I only see it in `Force.GetBatchInfo`, where we will get back JSON if we request the batch info for a valid JSON job and invalid batch, for example:
```
{
    "exceptionCode": "InvalidBatch",
    "exceptionMessage": "Unable to find batch for id: xxxxx"
}
```